### PR TITLE
CSS Reshuffle

### PR DESCRIPTION
**Pull request by mateoclarke**
_Closes cityofaustin/techstack#272_

All component js and scss are subfoldered under `src/components/ExampleComponentName`.

Groupings of related components are in folders like: 
- `src/components/Pages`
- `src/components/PageSections`
- `src/components/Tiles`

Open for ideas to make that more discoverable.

We could move even more components into PageSections. Next PR.

We could also probably name js files `index.js` instead of `ExampleComponentName.js` to make imports cleaner, but I hesitated to do that in this PR.  I didn't want to block other CSS improvements by making this PR even bigger.

### CODE REVIEW

File: src/css/coa.scss [hunk 1]
```diff
@@ -17,32 +17,30 @@
...
-@import "css/page_sections/Header";
-@import "css/page_sections/Footer";
-@import "css/page_sections/LanguageSelectBar";
-@import "css/page_sections/Menu/Menu";
-@import "css/page_sections/Menu/MenuItem";
-@import "css/page_sections/Menu/Submenu";
-@import "css/page_sections/Menu/SubmenuItem";
-@import 'css/page_sections/SecondaryContentBanner';
-@import 'css/page_sections/ThreeOneOne';
+// components
```
[the remark this answers is not in the document]
Comment (benweatherman) [reply src/css/coa.scss]: It might be weird to support globbing with SASS because of CSS override rules

[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/css/coa.scss]: Hmm, i never knew that was what you call it. Globbing, what a funny word.

[the remark this answers is not in the document]
Comment (ifsimicoded) [reply src/css/coa.scss]: screen shot 2018-04-24 at 10 28 31 pm

File: src/stories/Contact.stories.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import { storiesOf } from '@storybook/react';
 
-import ContactMap from 'js/modules/Contact/ContactMap';
-import ContactDetails from 'js/modules/Contact/ContactDetails';
+import ContactMap from 'components/Contact/ContactMap';
+import ContactDetails from 'components/Contact/ContactDetails';
 import ContactPhone from 'js/modules/Contact/Phone';
```
Comment (ifsimicoded) [inline src/stories/Contact.stories.js]: I think lines 6, 7, 8 the paths need to be updated too

**Comment by ifsimicoded** [comment]
whatcha think about moving svgs over too?

**Comment by ifsimicoded** [comment]
Are we also planning on killing the css manifest and doing style includes within the components?

File: src/components/Pages/Department.js [hunk 3]
```diff
@@ -3,11 +3,13 @@ import { get } from 'lodash';
 import { cleanContacts } from 'js/helpers/cleanData';
 import { withRouteData } from 'react-static';
 
+import SectionHeader from 'components/SectionHeader/SectionHeader';
+import PageHeader from 'components/PageHeader/PageHeader';
+import ContactDetails from 'components/Contact/ContactDetails';
+import ThreeOneOne from 'components/PageSections/ThreeOneOne/ThreeOneOne';
```
Comment (ifsimicoded) [inline src/components/Pages/Department.js]: ThreeOneOne got added here, not sure why.

File: src/components/Pages/Home.js [hunk 4]
```diff
@@ -2,12 +2,12 @@ import React from 'react';
...
 import { FormattedMessage, defineMessages, injectIntl } from 'react-intl';
 
-import SecondaryContentBanner from 'js/page_sections/SecondaryContentBanner';
-import HeroHome from 'js/modules/HeroHome';
-import ExternalLink from 'js/modules/ExternalLink';
-import SectionHeader from 'js/modules/SectionHeader';
-import TileGroup from 'js/modules/TileGroup';
-
+import SecondaryContentBanner from 'components/PageSections/SecondaryContentBanner/SecondaryContentBanner';
+import ThreeOneOne from 'components/PageSections/ThreeOneOne/ThreeOneOne';
```
Comment (ifsimicoded) [inline src/components/Pages/Home.js]: same, ThreeOneOne is in the footer now. probably a weird merge sitch.

File: src/components/Pages/Service.js [hunk 5]
```diff
@@ -3,16 +3,17 @@ import { withRouteData } from 'react-static';
...
+import PageBreadcrumbs from 'components/PageBreadcrumbs/PageBreadcrumbs';
+import PageHeader from 'components/PageHeader/PageHeader';
+import Steps from 'components/Steps/Steps';
+import HtmlFromAdmin from 'components/HtmlFromAdmin/HtmlFromAdmin';
+import ApplicationBlock from 'components/ApplicationBlock/ApplicationBlock';
+import ContactDetails from 'components/Contact/ContactDetails';
+import SectionHeader from 'components/SectionHeader/SectionHeader';
+import TileGroup from 'components/Tiles/TileGroup';
+import FormFeedback from 'components/FormFeedback/FormFeedback';
+import ThreeOneOne from 'components/PageSections/ThreeOneOne/ThreeOneOne';
```
Comment (ifsimicoded) [inline src/components/Pages/Service.js]: 311 ⛔️

File: src/components/Pages/Services.js [hunk 6]
```diff
@@ -2,8 +2,9 @@ import React from 'react';
 import { withRouteData } from 'react-static';
 import { defineMessages, injectIntl } from 'react-intl';
 
-import PageHeader from 'js/modules/PageHeader';
-import TileGroup from 'js/modules/TileGroup';
+import PageHeader from 'components/PageHeader/PageHeader';
+import TileGroup from 'components/Tiles/TileGroup';
+import ThreeOneOne from 'components/PageSections/ThreeOneOne/ThreeOneOne';
```
Comment (ifsimicoded) [inline src/components/Pages/Services.js]: 311 ⛔️

File: src/components/Pages/Theme.js [hunk 7]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import { withRouteData } from 'react-static';
 import PropTypes from 'prop-types';
 
-import TileGroupSet from 'js/modules/TileGroupSet';
-import PageHeader from 'js/modules/PageHeader';
-
+import TileGroupSet from 'components/Tiles/TileGroupSet';
+import PageHeader from 'components/PageHeader/PageHeader';
+import ThreeOneOne from 'components/PageSections/ThreeOneOne/ThreeOneOne';
```
Comment (ifsimicoded) [inline src/components/Pages/Theme.js]: 311 ⛔️

File: src/stories/ThreeOneOne.stories.js [hunk 8]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import { storiesOf } from '@storybook/react';
 
-import ThreeOneOne from 'js/page_sections/ThreeOneOne';
-import threeoneoneData from 'stories/static_data/threeoneoneData';
+import ThreeOneOne from 'components/PageSections/ThreeOneOne/ThreeOneOne';
+import jsonFileData from 'stories/static_data/pages';
```
Comment (ifsimicoded) [inline src/stories/ThreeOneOne.stories.js]: probably a merge 🌀-- I think pages on line 5 should be threeoneoneData

**Comment by mateoclarke** [comment]
@ifsimicoded:
- I followed your suggestion on moving SVGs.
- Also renamed solo component js files within a directory `index.js`
- I tried dropping the underscore and importing the CSS directly into the component js file. But I hit the issue of the Sass variables being undefined and choking the compile to css step. This would probably be easy to troubleshoot by knowing what files to import as a dependency but I remembered there was duplicate code issues with that in the past. And then I got lazy. Either way, we can do that in another PR. 

Gonna merge...

**Pull request merged**